cpp/modmesh/math/Complex.hpp

```diff
@@ -106,54 +111,114 @@ struct ComplexImpl
         return *this;
     }
 
-    ComplexImpl & operator+=(const ComplexImpl & other)
-    {
-        real_v += other.real_v;
-        imag_v += other.imag_v;
-        return *this;
-    }
-
-    ComplexImpl & operator-=(const ComplexImpl & other)
-    {
-        real_v -= other.real_v;
-        imag_v -= other.imag_v;
-        return *this;
-    }
-
-    bool operator<(const ComplexImpl<T> & rhs) const
-    {
-        return this->norm() < rhs.norm();
-    }
-
-    bool operator>(const ComplexImpl<T> & rhs) const
-    {
-        return this->norm() > rhs.norm();
-    }
-
     T real() const { return real_v; }
     T imag() const { return imag_v; }
     T norm() const { return real_v * real_v + imag_v * imag_v; }
     ComplexImpl<T> conj() const { return {real_v, -imag_v}; }
 }; /* end struct ComplexImpl */
 
-} /* end namespace detail */
-
-template <typename T>
-using Complex = detail::ComplexImpl<T>;
-
 // These comparison operator would be used in SimpleArray::min(), SimpleArray::max().
 template <typename T>
-bool operator<(const Complex<T> & lhs, const Complex<T> & rhs)
+bool operator<(const ComplexImpl<T> & lhs, const ComplexImpl<T> & rhs)
 {
     return lhs.norm() < rhs.norm();
 }
 
 template <typename T>
-bool operator>(const Complex<T> & lhs, const Complex<T> & rhs)
+bool operator>(const ComplexImpl<T> & lhs, const ComplexImpl<T> & rhs)
 {
     return lhs.norm() > rhs.norm();
 }
 
+template <typename T>
+bool operator==(const ComplexImpl<T> & lhs, const ComplexImpl<T> & rhs)
+{
+    return lhs.real_v == rhs.real_v && lhs.imag_v == rhs.imag_v;
+}
+
+template <typename T>
+bool operator!=(const ComplexImpl<T> & lhs, const ComplexImpl<T> & rhs)
+{
+    return lhs.real_v != rhs.real_v || lhs.imag_v != rhs.imag_v;
+}
+
+template <typename T>
+ComplexImpl<T> operator+(ComplexImpl<T> lhs, const ComplexImpl<T> & rhs)
+{
+    return lhs += rhs;
+}
+
+template <typename T>
+ComplexImpl<T> operator+(ComplexImpl<T> lhs, T rhs)
```

Comment:
Also add `complex @ real` and `real @ complex` for all the 4 arithmetic operators.  The real number $v$ is the complex number $v + 0j$.